fitbenchmarking/fitting/controllers/gsl_controller.py

```diff
@@ -0,0 +1,188 @@
+        self._pinit = numx.array(self.initial_params)
+
+        self._residual_methods = ['lmsder',
+                                  'lmder']
+        self._function_methods_no_jac = ['simplex',
+                                         'simplex2']
+        self._function_methods_with_jac = ['conjugate_pr',
+                                           'conjugate_fr',
+                                           'vector_bfgs',
+                                           'vector_bfgs2',
+                                           'steepest_descent']
+
+        # set up the system
+        if self.minimizer in self._residual_methods:
+            mysys = multifit_nlin.gsl_multifit_function_fdf(
+                self._prediction_error,
+                self._jac,
+                self._fdf,
+                self._data,
+                self._n,
+                self._p)
+        elif self.minimizer in self._function_methods_no_jac:
+            mysys = multiminimize.gsl_multimin_function(self._chi_squared,
+                                                        self._data,
+                                                        self._p)
+        elif self.minimizer in self._function_methods_with_jac:
+            mysys = multiminimize.gsl_multimin_function_fdf(self._chi_squared,
+                                                            self._jac_chi_squared,
+                                                            self._chi_squared_fdf,
+                                                            self._data,
+                                                            self._p)
+        else:
+            raise RuntimeError("An undefined GSL minimizer was selected")
+            
+        # define the solver
+        if self.minimizer=='lmsder':
+            self._solver = multifit_nlin.lmsder(mysys, self._n, self._p)
+        elif self.minimizer=='lmder':
+            self._solver = multifit_nlin.lmder(mysys, self._n, self._p)
+        elif self.minimizer=='simplex':
+            self._solver = multiminimize.nmsimplex(mysys, self._p)
+        elif self.minimizer=='simplex2':
+            self._solver = multiminimize.nmsimplex2(mysys, self._p)
+        elif self.minimizer=='conjugate_pr':
+            self._solver = multiminimize.conjugate_pr(mysys, self._p)
+        elif self.minimizer=='conjugate_fr':
+            self._solver = multiminimize.conjugate_fr(mysys, self._p)
+        elif self.minimizer=='bfgs':
+            self._solver = multiminimize.bfgs(mysys, self._p)
+        elif self.minimizer=='bfgs2':
+            self._solver = multiminimize.bfgs2(mysys, self._p)
+        elif self.minimizer=='steepest_descent':
+            self._solver = multiminimize.steepest_descent(mysys, self._p)
+
+        # Set up initialization parameters
+        # 
+        # These have been chosen to be consistent with the parameters
+        # used in Mantid.
+        self._initial_steps = 1.0 * numx.array(np.ones(self._p))
```

Comment:
These params could either be defined in the `__init__` method or they do not need to be defined on self.